### PR TITLE
Optimise docker image build layers

### PR DESCRIPTION
**Pull request by slyg**
Resolves [CNP-1109](https://tools.hmcts.net/jira/browse/CNP-1109)

Hello, this PR adds the following enhancements:
- improve the ordering of the docker image build layer 
- prepare for future ACR cache optimisations.

This is part of a general [pipelines optimisations initiative](https://tools.hmcts.net/confluence/display/CNP/Pipeline+optimisations) and should not impact the current builds.

### CODE REVIEW

**Comment by slyg** [comment]
Hi @markry-kainos indeed there was an issue now solved, but as we are not sure about the direction we might take afterward regarding production. As these images are only used for tests, it is probably safe to keep this node-slim image, not too big but still with a few cli commands ready in case you need to debug.

We'll provide a best practices guide on confluence to keep track of this. 

Thx !

File: Dockerfile [hunk 1]
```diff
@@ -1,21 +1,24 @@
-FROM node:8.12.0-slim
-
-RUN mkdir -p /usr/src/app
-WORKDIR /usr/src/app
-
-COPY package.json yarn.lock tsconfig.json tsconfig.prod.json gulpfile.js /usr/src/app/
-COPY src/main /usr/src/app/src/main
-
+# ---- Base image ----
+FROM node:8.12.0-slim as base
```
Comment (timja) [inline Dockerfile]: alpine is fine to use now would be smaller but doesn't really matter

Comment (slyg) [reply Dockerfile]: Yup you are right Tim, though we didn't want to change too much regarding the base images as we are still assessing the best choices. As suggested by @timwebster9 some base images can be small and AKS-friendly but until then, we think it is good enough :)

Thx mate !

**Comment by slyg** [comment]
Hi, seems this project fails to pass the smoke tests on all branches, is it worth keeping this PR opened ?

**Comment by markry-kainos** [comment]
> Hi, seems this project fails to pass the smoke tests on all branches, is it worth keeping this PR opened ?

I will take a look next week @slyg

**Comment by slyg** [comment]
Cheer @markry-kainos

**Comment by slyg** [comment]
Thx 👍🏼

**Pull request merged**